### PR TITLE
Set Functional property after battery CM

### PR DESCRIPTION
**Pull request by spinler**
When bmcweb is told the RTC battery is replaced after a CM operation, set the Functional property on its assembly object path back to true before restarting the adcsensor daemon.

This does two things:
1) Causes the fault LED to turn off
2) Sets the Redfish representation back to healthy so the web UI doesn't
   show it has unhealthy it after the CM operation.  Of course if
   adcsensor detects an error again it would go back to unhealthy.

Since this is handling the fault LED, the previous code that explicitly turned off the fault LED group was removed.

Defect PE00DX9W.  For 1030.10

Signed-off-by: Matt Spinler <spinler@us.ibm.com>

### CODE REVIEW

File: redfish-core/lib/assembly.hpp [hunk 1]
```diff
@@ -435,6 +435,96 @@ inline void
         "xyz.openbmc_project.Association", "endpoints");
 }
 
+void startOrStopADCSensor(bool start,
```
Comment (sbteeks) [inline redfish-core/lib/assembly.hpp:438]: Maybe 'const bool start'?

File: redfish-core/lib/assembly.hpp [hunk 2]
```diff
@@ -435,6 +435,96 @@ inline void
...
+                return;
+            }
+            messages::success(asyncResp->res);
+        },
+        "org.freedesktop.systemd1", "/org/freedesktop/systemd1",
+        "org.freedesktop.systemd1.Manager", method,
+        "xyz.openbmc_project.adcsensor.service", "replace");
+}
+
+void doBatteryCM(const std::string& assembly, bool readyToRemove,
```
Comment (sbteeks) [inline redfish-core/lib/assembly.hpp:462]: 'const bool'?

Comment (spinler) [reply redfish-core/lib/assembly.hpp]: done

**Comment by gtmills** [comment]
```
clang-tidy-14 --use-color -export-fixes /tmp/tmpwiyfiv0u/tmpe4s0l3ek.yaml -p=. /home/fspcibld/openbmc-dev-jenkins/workspace/OpenBMC-Dev/build/github-public/repo-ci/public-ibm-openbmc-downstream-repo-ci-pr-bmcweb/ibm-openbmc/bmcweb/src/webserver_main.cpp
[1m../redfish-core/include/../lib/assembly.hpp:496:33: [0m[0;1;31merror: [0m[1m'serviceName' in capture list does not name a variable [clang-diagnostic-error][0m
                    [asyncResp, serviceName,
[0;1;32m                                ^
[0m197736 warnings and 1 error generated.
```
is the error

File: redfish-core/lib/assembly.hpp [hunk 3]
```diff
@@ -435,6 +435,96 @@ inline void
...
+                 const std::shared_ptr<bmcweb::AsyncResp>& asyncResp)
+{
+    if (readyToRemove)
+    {
+        // Stop the adcsensor service so it doesn't monitor the battery
+        startOrStopADCSensor(false, asyncResp);
+        return;
+    }
+
+    // Find the service that has the OperationalStatus iface, set the
```
Comment (gtmills) [inline redfish-core/lib/assembly.hpp:472]: This comment is helpful.

When readyToRemove is true, we are stopping the adcsensor, that code is above on line 468. Makes sense. 

When readyToRemove is false, this is the only time we get in here. 
readyToRemove =  false is done after the CM.. 
So when readyToRemove is false, we set functional to true and start the ADC sensor. Makes sense.

**Review by gtmills — APPROVED** [review]
I am good once it passes CI

**Comment by spinler** [comment]
> Huh, I don't see how that's an error. Also, it works fine.

yea that is an error .fixed.

**Review by sbteeks — APPROVED** [review]

**Pull request merged**